contracts/shiden34/lib.rs

```diff
@@ -67,9 +62,44 @@ pub mod shiden34 {
         });
     }
 
-    impl payable_mint_pkg::impls::payable_mint::payable_mint::Internal for Shiden34 {}
+    impl payable_mint::Internal for Shiden34 {}
+
     impl payable_mint::PayableMintImpl for Shiden34 {}
 
+    impl PayableMint for Shiden34 {
+        #[ink(message, payable)]
+        fn mint(&mut self, to: AccountId, mint_amount: u64) -> Result<(), PSP34Error> {
+            payable_mint::PayableMintImpl::mint(self, to, mint_amount)
```

Comment:
I do not think this is a good way.  You have 2 implementation of basically the same trait
It should be as simple as 
```

impl payable_mint::PayableMint for Shiden34Contract {}
```

Reply:
This would be ideal. I will investigate how to do that with the latest Openbrush